x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+    private final String error;
+    @Nullable
+    private final ConnectorFeatures features;
+
+    // TODO: it's a pretty complex object, it will come as a separate PR
+    // private final List<ConnectorFilteringConfig> filtering;
+
+    @Nullable
+    private final String indexName;
+    @Nullable
+    private final boolean isNative;
+    @Nullable
+    private final String language;
+    @Nullable
+    private final String lastAccessControlSyncError;
+    @Nullable
+    private final String lastAccessControlSyncScheduledAt;
+    @Nullable
+    private final ConnectorSyncStatus lastAccessControlSyncStatus;
+    @Nullable
+    private final Long lastDeletedDocumentCount;
+    @Nullable
+    private final String lastIncrementalSyncScheduledAt;
+    @Nullable
+    private final Long lastIndexedDocumentCount;
+    @Nullable
+    private final String lastSeen;
+    @Nullable
+    private final String lastSyncError;
+    @Nullable
+    private final String lastSyncScheduledAt;
+    @Nullable
+    private final ConnectorSyncStatus lastSyncStatus;
+    @Nullable
+    private final String lastSynced;
+    @Nullable
+    private final String name;
+    @Nullable
+    private final ConnectorIngestPipeline pipeline;
+    @Nullable
+    private final ConnectorScheduling scheduling;
+    @Nullable
+    private final String serviceType;
+    @Nullable
+    private final ConnectorStatus status;
+    @Nullable
+    private final Object syncCursor;
+    @Nullable
+    private final boolean syncNow;
+
+    private Connector(
+        String connectorId,
+        String apiKeyId,
+        Map<String, Object> configuration,
+        ConnectorCustomSchedule customScheduling,
+        String description,
+        String error,
+        ConnectorFeatures features,
+        String indexName,
+        boolean isNative,
+        String language,
+        String lastAccessControlSyncError,
+        String lastAccessControlSyncScheduledAt,
+        ConnectorSyncStatus lastAccessControlSyncStatus,
+        Long lastDeletedDocumentCount,
+        String lastIncrementalSyncScheduledAt,
+        Long lastIndexedDocumentCount,
+        String lastSeen,
+        String lastSyncError,
+        String lastSyncScheduledAt,
+        ConnectorSyncStatus lastSyncStatus,
+        String lastSynced,
+        String name,
+        ConnectorIngestPipeline pipeline,
+        ConnectorScheduling scheduling,
+        String serviceType,
+        ConnectorStatus status,
+        Object syncCursor,
+        boolean syncNow
+    ) {
+        this.connectorId = connectorId;
+        this.apiKeyId = apiKeyId;
+        this.configuration = configuration;
+        this.customScheduling = customScheduling;
+        this.description = description;
+        this.error = error;
+        this.features = features;
+        this.indexName = indexName;
+        this.isNative = isNative;
+        this.language = language;
+        this.lastAccessControlSyncError = lastAccessControlSyncError;
```

Comment:
I wonder, would it make sense to group some of these fields into classes? We can serialize them as a non-nested JSON object, but it would maybe simplify code and serialization management. I can think of creating a SyncStatus class that would contain:

        String lastAccessControlSyncError,
        String lastAccessControlSyncScheduledAt,
        Long lastDeletedDocumentCount,
      String lastSyncError,
        String lastSyncScheduledAt,
        ConnectorSyncStatus lastSyncStatus,
        String lastSynced,
        Object syncCursor,
        boolean syncNow
        String lastSyncError,
        String lastSyncScheduledAt,
        Long lastIndexedDocumentCount,

Reply:
Yeah I was actually thinking about the same when I was looking at the `Connector` class. I didn't realise that we can serialise this as as non-nested structure. 👍  That will simplify things a bit :)